spec/features/idv/analytics_spec.rb

```diff
@@ -134,7 +134,10 @@
   end
 
   before do
-    allow_any_instance_of(ApplicationController).to receive(:analytics).and_return(fake_analytics)
+    allow_any_instance_of(ApplicationController).to receive(:analytics) do |controller|
+      fake_analytics.user = controller.analytics_user
+      fake_analytics
+    end
```

Comment:
This feels rather hacky, but I was finding that the specs were failing due to what appeared to be the user record being outdated (`user.proofing_component` vs. `user.reload.proofing_component` had different results when using `binding.pry`). I suspected it's because the stubbed `user` was being cached across requests, which is unlike how it would behave in the real-world. Interesting that it only happened when updating proofing component for `gpo_letter` (pretty late in the flow) and not other proofing components.